### PR TITLE
WIP: minimal+base+sdk+proxy_SCC-postreg for SLE15

### PR DESCRIPTION
**Pull request by yxususe**
Approach for ticket: https://progress.opensuse.org/issues/19566

### CODE REVIEW

**Comment by yxususe** [comment]
Local testrun: f146.suse.de/tests/1016

**Comment by okurz** [comment]
need to tidy

**Comment by okurz** [comment]
the test did not get far. You need to rebase to get support for the new product dialog. Also an updated needles repo is suggested

File: products/sle/main.pm [hunk 1]
```diff
@@ -758,7 +758,12 @@ sub load_consoletests {
     }
     #have SCC repo for SLE product
     if (have_scc_repos()) {
-        loadtest "console/yast_scc";
+        if (sle_version_at_least '15') {
+            loadtest "console/proxyscc_postreg";
```
Comment (okurz) [inline products/sle/main.pm]: SUSEConnect was already there in before. I don't see a need to that as an alternative. How about specifying a specific test variable for this mode?

File: tests/console/proxyscc_postreg.pm [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2014-2017 SUSE LLC
```
Comment (okurz) [inline tests/console/proxyscc_postreg.pm]: just 2017

File: tests/console/proxyscc_postreg.pm [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# Maintainer: Yi Xu <yxu@suse.de>
+
+use strict;
+use base "y2logsstep";
+
+use testapi;
+use registration;
+
+sub run {
+    if (!get_var("HDD_SCC_REGISTERED")) {
```
Comment (okurz) [inline tests/console/proxyscc_postreg.pm]: Just use an early exit, e.g.

```perl
return if get_var("HDD_SCC_REGISTERED");
```

File: tests/console/proxyscc_postreg.pm [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+	# add repo
+        script_run "zypper ar -f http://download.suse.de/ibs/SUSE:/SLE-15:/GA/standard/SUSE:SLE-15:GA.repo";
```
Comment (okurz) [inline tests/console/proxyscc_postreg.pm]: Don't use blank script_run without checking for the exit code. Better always use `assert_script_run` by default. But actually we have a utility function `utils::zypper_call` which you can use.

Also, do not add an IBS repo without a `record_soft_failure`

File: tests/console/proxyscc_postreg.pm [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        script_run "zypper in sles-release";
+        script_run "zypper rr 1";
+
+	# fix simlink
+	script_run "rm /etc/products.d/baseproduct";
+	script_run "ln -s SLES.prod /etc/products.d/baseproduct";
+
+	# activate base SLES15
+	assert_script_run "SUSEConnect -r 30452ce234918d23";
```
Comment (okurz) [inline tests/console/proxyscc_postreg.pm]: please don't put regcodes out in the open. We had this already.

File: tests/console/proxyscc_postreg.pm [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	script_run "SUSEConnect --list-extensions";
+
+	# add modules
+	assert_script_run "SUSEConnect -p sle-module-basesystem/15/x86_64";
```
Comment (okurz) [inline tests/console/proxyscc_postreg.pm]: better use the portion from the version variable

File: tests/console/proxyscc_postreg.pm [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+	assert_script_run "SUSEConnect -p sle-module-scripting/15/x86_64";
+	assert_script_run "SUSEConnect -p sle-module-desktop-applications/15/x86_64";
```
Comment (rwx788) [inline tests/console/proxyscc_postreg.pm]: Please, make test architecture independent. Yes, now it works only on x86-64, but will work on other arch in future. The same states with version.

File: tests/console/proxyscc_postreg.pm [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+use testapi;
+use registration;
+
+sub run {
+    if (!get_var("HDD_SCC_REGISTERED")) {
+
+	# add repo
+        script_run "zypper ar -f http://download.suse.de/ibs/SUSE:/SLE-15:/GA/standard/SUSE:SLE-15:GA.repo";
+        script_run "zypper in sles-release";
```
Comment (rwx788) [inline tests/console/proxyscc_postreg.pm]: Please, verify if this is still needed, as we can select SLES as product during installation, so these workarounds may not be required.

**Comment by okurz** [comment]
closed to hide the regcode at least a little bit

File: products/sle/main.pm (hunk 1 above)
Comment (yxususe) [reply products/sle/main.pm]: yast_scc is the way how it was done before, from sle15 we do it differently, isn't it?

File: tests/console/proxyscc_postreg.pm (hunk 2 above)
Comment (yxususe) [reply tests/console/proxyscc_postreg.pm]: done.

File: tests/console/proxyscc_postreg.pm (hunk 3 above)
Comment (yxususe) [reply tests/console/proxyscc_postreg.pm]: done.

File: tests/console/proxyscc_postreg.pm (hunk 7 above)
Comment (yxususe) [reply tests/console/proxyscc_postreg.pm]: Done.

File: tests/console/proxyscc_postreg.pm (hunk 8 above)
Comment (yxususe) [reply tests/console/proxyscc_postreg.pm]: Still trying to get it run.

File: products/sle/main.pm (hunk 1 above)
Comment (okurz) [reply products/sle/main.pm]: I don't see a reason how sle15 should behave different regarding the use of SUSEConnect except for the list of modules

**Pull request closed without merging**